notebooks/search/06-synonymys-api.ipynb

```diff
@@ -0,0 +1,517 @@
+        "\n",
+        "Now let's try searching for books about AI."
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "colab": {
+          "base_uri": "https://localhost:8080/"
+        },
+        "id": "oj1ynL5nZz0u",
+        "outputId": "f1968d2c-83a5-4b3c-f397-44b16e7ab46e"
+      },
+      "outputs": [],
+      "source": [
+        "response = client.search(\n",
+        "    index=\"book_index\",\n",
+        "    query={\n",
+        "        \"multi_match\": {\n",
+        "            \"query\": \"AI\",\n",
+        "            \"fields\": [\n",
+        "                \"title^10\",\n",
+        "                \"summary\",\n",
+        "            ]\n",
+        "        }\n",
+        "    }\n",
+        ")\n",
+        "\n",
+        "pretty_search_response(response)"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "RtXj_JwyZ3DZ"
+      },
+      "source": [
+        "We don't get any results! Let's try using the Synonyms API to add a new synonym rule for AI.\n",
+        "\n"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "colab": {
+          "base_uri": "https://localhost:8080/"
+        },
+        "id": "8sZ4nkpzgwMy",
+        "outputId": "d425906a-3f6e-4dc2-89ed-ca6bbef70b0b"
+      },
+      "outputs": [],
+      "source": [
+        "client.synonyms.put_synonym_rule(set_id=\"my-synonyms-set\", rule_id=\"synonym-2\", synonyms=\"ai, artificial intelligence\")"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "KFgKAma1hMT_"
+      },
+      "source": [
+        "If we run the query again, we should now get some results."
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "colab": {
+          "base_uri": "https://localhost:8080/"
+        },
+        "id": "KDx_V__QhIiy",
+        "outputId": "6d23e7f1-e129-4ee7-edf7-8e55ba1d0355"
+      },
+      "outputs": [],
+      "source": [
+        "response = client.search(\n",
+        "    index=\"book_index\",\n",
+        "    query={\n",
+        "        \"multi_match\": {\n",
+        "            \"query\": \"AI\",\n",
+        "            \"fields\": [\n",
+        "                \"title^10\",\n",
+        "                \"summary\",\n",
+        "            ]\n",
+        "        }\n",
+        "    }\n",
+        ")\n",
+        "\n",
+        "pretty_search_response(response)"
+      ]
+    }
+  ],
+  "metadata": {
+    "colab": {
+      "provenance": []
+    },
+    "kernelspec": {
+      "display_name": "Python 3",
+      "name": "python3"
+    },
+    "language_info": {
+      "name": "python"
+    }
+  },
+  "nbformat": 4,
```

Comment:
Should we add some conclusion like sentences like "Using synonyms API allows you to..."? I'm including search experiences team to review as well as @kderusso mentioned